src/Tasks.UnitTests/CodeTaskFactoryTests.cs

```diff
@@ -1040,6 +1043,164 @@ public override bool Execute()
             }
         }
 
+        [Fact]
+        public void EmbedsSourceFileInBinlog()
+        {
+            string sourceFileContent = @"
+                                       using System;
+                    using System.Collections.Generic;
+                    using System.Text;
+                    using Microsoft.Build.Utilities;
+                    using Microsoft.Build.Framework;
+
+                    namespace Microsoft.Build.NonShippingTasks
+                    {
+                        public class LogNameValue_ClassSourcesTest : Task
+                        {
+                            private string variableName;
+                            private string variableValue;
+
+
+                            [Required]
+                            public string Name
+                            {
+                                get { return variableName; }
+                                set { variableName = value; }
+                            }
+
+
+                            public string Value
+                            {
+                                get { return variableValue; }
+                                set { variableValue = value; }
+                            }
+
+
+                            public override bool Execute()
+                            {
+                                // Set the process environment
+                                Log.LogMessage(""Setting {0}={1}"", this.variableName, this.variableValue);
+                                return true;
+                            }
+                        }
+                    }
+";
+
+            string tempFileDirectory = Path.GetTempPath();
+            string tempFileName = Guid.NewGuid().ToString() + ".cs";
+            string tempSourceFile = Path.Combine(tempFileDirectory, tempFileName);
+            File.WriteAllText(tempSourceFile, sourceFileContent);
```

Comment:
We have helpers to keep all this sort of thing cleaned up and easy. Pushing a commit to use them here.

Reply:
Oh, I see you used them in the other class. While I like consistency within a file, I'd rather use the new/good stuff.

Reply:
Thanks! I will open a PR to cleanup `CodeTaskFactoryTests.cs`